management-app/src/api/usersAPI.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable no-console */
+import { BaseAPI } from './baseAPI';
+import { USERS_API_ERRORS } from './errors';
+import { TUser, TUserBase } from './types';
+
+class UsersAPI extends BaseAPI {
+  updateUser(
+    userId: string,
+    token: string,
+    dataUser: TUser,
+    updateUserCb: (data: TUser) => void,
+    errorCb = (_error: Error) => {
+      console.error(_error);
+    }
+  ) {
+    this.put(`users/${userId}`, dataUser, {
+      Authorization: `Bearer ${token}`,
+    })
+      .then((result) => {
+        BaseAPI.handleError(result, USERS_API_ERRORS);
+        return result.json();
+      })
+      .then((result) => updateUserCb(result))
+      .catch((error) => errorCb(error));
+  }
+
+  getUser(
+    userId: string,
+    token: string,
+    getUserCb: (data: TUser) => void,
```

Comment:
what do you think about using promises instead of callbacks?

Reply:
What do you mean? Could you code example?

Reply:
@MarinaPresmytskaia could you please check this example: https://github.com/treyer/project-management-app/pull/3/files#r867374863 ?

Reply:
I got it. Seams this approach will be better. Especially  if we will handle errors inside Redux thunk.